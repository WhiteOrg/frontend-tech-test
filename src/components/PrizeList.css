
.prize-list {
    padding-bottom: 0;
}

.prize-list h2 {
    display: flex;
    justify-content: space-between;
}

.score-board {
    font-family: "Barlow Condensed", sans-serif;
    font-size: 22px;
    font-weight: 600;
    border-top: 5px solid #F3F3FD ;
    border-bottom: 5px solid #F3F3FD ;
    padding: 25px 0;
}

.score-board ol {
    list-style-type: none;
    padding: 0;
    margin: 0;
    overflow: scroll;
    max-height: 310px;
    -webkit-overflow-scrolling: touch; /* Lets it scroll lazy */
}

.score-board li {
    border-bottom: 2px solid #F3F3FD ;
    padding: 12px 0 12px 13px;
    display: flex;
}

.score-board li:last-child {
    border-bottom: none;
}

.score-board .total {
    margin-left: auto;
}

.score-board .rank {
    width: 65px
}

.refresh {
    border:none;
    background-color:#001C62;
    color:#fff;
    width: 30px;
    height: 30px;
    border-radius: 50%;
    display: flex;
    align-items: center;
    justify-content: center;
}

/* .refresh svg {
    animation: App-logo-spin infinite 5s linear;
} */
/* 
@keyframes App-logo-spin {
	from {
		transform: rotate(0deg);
	}
	to {
		transform: rotate(360deg);
	}
} */