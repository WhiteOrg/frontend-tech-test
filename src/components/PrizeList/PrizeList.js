import React from "react";
import {
  Container,
  Heading,
  Content,
  ListContainer,
  ListItem,
  Index,
  Title,
  Prize,
} from "./PrizeList.styles";
import refresh from "../../images/refresh.svg";
import SectionTitle from "../SectionTitle/SectionTitle";

const PrizeList = ({ data, refreshData, isLoading, error }) => {
  const convertCurrencyToNumber = (data) => {
    if (data) {
      const convertedArray = data.map((item) => ({
        title: item.title,
        total: item.total,
        sortingIndex: parseFloat(item.total.replace(/[Â£,]/g, "")),
      }));
      return convertedArray;
    }
  };

  const sortData = (unsortedData) => {
    if (unsortedData) {
      const sortedData = unsortedData.sort(
        (a, b) => b.sortingIndex - a.sortingIndex
      );
      return sortedData;
    }
  };

  const convertedData = convertCurrencyToNumber(data);
  const finalData = sortData(convertedData);

  return (
    <Container>
      <Heading>
        <SectionTitle title="Prize List" />
        <img onClick={refreshData} src={refresh} alt="refresh-icon" />
      </Heading>
      <Content>
        <ListContainer>
          {error && <div>Couldn't fetch data</div>}
          {isLoading ? (
            <div>Refreshing...</div>
          ) : (
            finalData &&
            finalData.map((prize, index) => (
              <ListItem key={index}>
                <Index>{index + 1}</Index>
                <Title>{prize.title}</Title>
                <Prize>{prize.total}</Prize>
              </ListItem>
            ))
          )}
        </ListContainer>
      </Content>
    </Container>
  );
};

export default PrizeList;
