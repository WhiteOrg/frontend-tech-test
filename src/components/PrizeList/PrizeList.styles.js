import styled from "styled-components";

export const Container = styled.section`
  padding: 1.8rem;

  @media ${({ theme }) => theme.mediaQueries["tablet"]} {
    padding: 1.5rem 3rem;
  }

  @media ${({ theme }) => theme.mediaQueries["laptop"]} {
    padding: 2rem 4rem;
  }
  @media ${({ theme }) => theme.mediaQueries["desktop"]} {
    padding: 3rem 6rem;
  } ;
`;
export const Heading = styled.div`
  display: flex;
  width: 100%;
  justify-content: space-between;
  margin-bottom: 20px;

  > img {
    @media ${({ theme }) => theme.mediaQueries["tablet"]} {
      height: 25px;
      width: 25px;
    }
    @media ${({ theme }) => theme.mediaQueries["laptop"]} {
      height: 30px;
      width: 30px;
    }
    @media ${({ theme }) => theme.mediaQueries["desktop"]} {
      height: 40px;
      width: 40px;
    }
    &:hover {
      cursor: pointer;
    }
  }
`;

export const Content = styled.div`
  border-top: 4px solid #f3f2fd;
  border-bottom: 4px solid #f3f2fd;
  padding: 20px 0;
  @media ${({ theme }) => theme.mediaQueries["tablet"]} {
    border-top: 4.5px solid #f3f2fd;
    border-bottom: 4.5px solid #f3f2fd;
  }
  @media ${({ theme }) => theme.mediaQueries["laptop"]} {
    border-top: 5px solid #f3f2fd;
    border-bottom: 5px solid #f3f2fd;
  }
  @media ${({ theme }) => theme.mediaQueries["desktop"]} {
    border-top: 6px solid #f3f2fd;
    border-bottom: 6px solid #f3f2fd;
  }
`;

export const ListContainer = styled.div`
  width: 100%;
  height: calc((6 * 35px) - 1px);
  overflow-y: scroll;
  position: relative;
  @media ${({ theme }) => theme.mediaQueries["tablet"]} {
    height: calc((6 * 40px) - 1px);
  }

  @media ${({ theme }) => theme.mediaQueries["laptop"]} {
    height: calc((6 * 45px) - 1px);
  }

  @media ${({ theme }) => theme.mediaQueries["desktop"]} {
    height: calc((6 * 50px) - 1px);
  }

  &::-webkit-scrollbar {
    width: 4px;
    @media ${({ theme }) => theme.mediaQueries["tablet"]} {
      width: 4.5px;
    }
    @media ${({ theme }) => theme.mediaQueries["laptop"]} {
      width: 5px;
    }
    @media ${({ theme }) => theme.mediaQueries["desktop"]} {
      width: 6px;
    }
  }
  &::-webkit-scrollbar-thumb {
    background-color: #f9cb11;
    border-radius: 10px;
  }
  &::-webkit-scrollbar-track {
    background-color: #f3f2fd;
  }
`;

export const ListItem = styled.li`
  font-family: ${({ theme }) => theme.fonts.primary};
  font-size: 1rem;
  height: 35px;
  font-weight: 600;
  display: flex;
  align-items: center;
  border-bottom: 1px solid #f3f2fd;
  margin-right: 1.5rem;
  @media ${({ theme }) => theme.mediaQueries["tablet"]} {
    font-size: 1.2rem;
    height: 40px;
  }
  @media ${({ theme }) => theme.mediaQueries["laptop"]} {
    font-size: 1.4rem;
    height: 45px;
  }
  @media ${({ theme }) => theme.mediaQueries["desktop"]} {
    font-size: 1.6rem;
    height: 50px;
  }
`;

export const Index = styled.span`
  display: block;
  width: 50px;
`;
export const Title = styled.span`
  display: block;
`;
export const Prize = styled.span`
  display: block;
  flex: 1;
  text-align: right;
`;
