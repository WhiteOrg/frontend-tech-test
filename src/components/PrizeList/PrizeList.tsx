import React from 'react'
import styled from 'styled-components/macro'
import data from '../../data.json'
import { COLORS, WEIGHTS } from '../../constants'
import refresh from '../../images/refresh.svg'

const PrizeList: React.FC = () => (
  <Wrapper>
    <TitleWrapper>
      <PrizeListTitle>Prize List</PrizeListTitle>
      <RefreshButton tabIndex={0} src={refresh} alt="refresh" />
    </TitleWrapper>
    <Divider />
    <ListWrapper>
      {data.prizeList.map((prize, index) => (
        <ListItem key={prize.title}>
          <ListItemIndex>{index}</ListItemIndex>
          <ListItemTitle>{prize.title}</ListItemTitle>
          <ListItemText>{prize.total}</ListItemText>
        </ListItem>
      ))}
    </ListWrapper>
    <Divider />
  </Wrapper>
)

const ListItemText = styled.p`
  font-size: 16px;
  font-weight: 600;
  font-stretch: condensed;
  font-style: normal;
  line-height: 2.06;
  letter-spacing: normal;
  color: ${COLORS.black};
`

const ListItemTitle = styled(ListItemText)`
  flex: 1;
`

const ListItemIndex = styled(ListItemText)`
  margin-right: 34px;
`

const ListItem = styled.li`
  display: flex;
  border-bottom: 1px solid ${COLORS.divider};
`

const ListWrapper = styled.ul`
  padding: 0px 16px 0 9px;
  margin: 20px 0;
  height: 198px;
  overflow: auto;

  ::-webkit-scrollbar {
    width: 4px;
  }

  ::-webkit-scrollbar-track {
    border-radius: 2px;
    background: ${COLORS.divider};
  }

  ::-webkit-scrollbar-thumb {
    border-radius: 2px;
    background: ${COLORS.secondary.main};
  }

  ::-webkit-scrollbar-thumb:hover {
    background: ${COLORS.secondary.light};
  }
`

const Divider = styled.div`
  background: ${COLORS.divider};
  height: 4px;
`

const TitleWrapper = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin-bottom: 17px;
`

const RefreshButton = styled.img`
  &:hover {
    cursor: pointer;
  }
`

const PrizeListTitle = styled.h2`
  font-size: 24px;
  font-stretch: condensed;
  font-style: normal;
  line-height: 1.08;
  letter-spacing: normal;
  color: ${COLORS.sectionTitle};
  font-weight: ${WEIGHTS.medium};
`

const Wrapper = styled.section`
  margin: 30px 21px 30px 20px;

  @media (min-width: 600px) {
    margin: 38px 72px 45px 68px;
  }
`

export default PrizeList
