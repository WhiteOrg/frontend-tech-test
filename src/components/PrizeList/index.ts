export { default as PrizeList } from './PrizeList'
