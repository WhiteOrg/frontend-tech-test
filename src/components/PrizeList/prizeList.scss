.prizeList {
  display: flex;
  padding: 0 20px;
  flex-direction: column;
  @media (min-width: 800px) {
    padding: 0 5%;
  }
  &__heading {
    display: flex;
    margin: 30px 0 20px;
    justify-content: space-between;
    align-items: center;
    h2 {
      margin: 0;
    }
    button {
      background: none;
      border: none;
      padding: 0;
      font: inherit;
      cursor: pointer;
      height: fit-content;
    }
  }
  table {
    width: 100%;
    margin: 20px 0;
  }
  &__table {
    display: flex;
    border-top: 2px solid rgba(128, 128, 128, 0.488);
    border-bottom: 2px solid rgba(128, 128, 128, 0.488);
    height: 235px;
    overflow-y: scroll;
    &--loading {
      display: flex;
      margin: auto;
    }
    tr {
      height: 32px;
      border-bottom: solid 2px rgba(128, 128, 128, 0.921);
      font-size: 16px;
      font-weight: 600;
      font-stretch: condensed;
      font-style: normal;
      line-height: 2.06;
      letter-spacing: normal;
      color: #000;
    }
    &--index {
      position: relative;
    }
    &--title {
      position: relative;
    }
    &--total {
      text-align: end;
      padding-right: 16px;
    }
  }
}

::-webkit-scrollbar-thumb {
  background: #f9cb11;
  border-radius: 2px;
}
::-webkit-scrollbar {
  width: 4px;
}
