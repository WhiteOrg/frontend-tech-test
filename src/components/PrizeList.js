import React from 'react';
import { connect } from 'react-redux';
import { fetchGames } from '../actions'
import { FaSync } from 'react-icons/fa'
import './PrizeList.css'

const PrizeList = ({ prizes, fetchGames }) => {

    const renderList = () => {
        return prizes.map((prize, index) => {
            return <li key={prize.title}>
                <span className="rank">{index + 1}</span>
                <span className="name">{prize.title}</span>
                <span className="total">{prize.total}</span>
            </li>
        })
    }

    const onRefresh = event => {
        event.preventDefault();
        fetchGames();
    }

    return (
        <section className="prize-list">
            <h2>Prize List <button className="refresh" onClick={onRefresh}><FaSync /></button></h2>
            <div className="score-board">
                <ol>
                    {(prizes) ? renderList() : null}
                </ol>
            </div>
        </section>
    );

}

const mapStateToProps = state => {
    return {
        prizes: state.game.tournament.prizeList
    }
}

export default connect(mapStateToProps, { fetchGames })(PrizeList);