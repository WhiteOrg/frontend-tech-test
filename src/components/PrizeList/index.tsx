import React, { useState } from "react";
import LoadingSpinner from "../LoadingSpinner";
import { getData } from "../../fetchData";
import { TPrizeList } from "../../utils/dataTypes";
import refresh from "../../images/refresh.svg";
import "./prizeList.scss";

const PrizeList = (props: { list: TPrizeList }) => {
  const { list } = props;
  const [isLoading, setIsLoading] = useState(false);
  const [listData, setListData] = useState(list);

  const refreshList = () => {
    setIsLoading(true);
    getData()
      .then((response) => response.json())
      .then((data) => {
        setListData(data.prizeList);
        setIsLoading(false);
      })
      .catch((error) => console.log("ERROR: ", error));
  };

  return (
    <div className="prizeList">
      <div className="prizeList__heading" data-testid="prize-heading">
        <h2>Prize List</h2>
        <button
          className="prizeList__heading--refresh"
          data-testid="refresh-button"
          onClick={refreshList}
        >
          <img src={refresh} alt="refresh icon" />
        </button>
      </div>

      <div className="prizeList__table">
        {isLoading ? (
          <div className="prizeList__table--loading">
            <LoadingSpinner />
          </div>
        ) : (
          <table data-testid="table-elem">
            <tbody>
              {listData.map((prize, index) => (
                <tr key={index} data-testid="table-row">
                  <td className="prizeList__table--index">{index + 1}</td>
                  <td className="prizeList__table--title">{prize.title}</td>
                  <td className="prizeList__table--total">{prize.total}</td>
                </tr>
              ))}
            </tbody>
          </table>
        )}
      </div>
    </div>
  );
};

export default PrizeList;
