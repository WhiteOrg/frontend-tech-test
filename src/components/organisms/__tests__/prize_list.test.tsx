import React from "react";
import renderer from "react-test-renderer";
import PrizeList from "../prize_list";
import refresh from "../../../images/refresh.svg";

describe("ORGANISMS: Render testing", () => {
  test("Should render PrizeList", () => {
    const data = {
      titlePrizeList: "Prize List",
      prizeList: [
        {
          title: "P.C",
          total: "£2,500.00",
        },
        {
          title: "J.S",
          total: "£1,250.00",
        },
        {
          title: "H.J",
          total: "£700.00",
        },
        {
          title: "J.V",
          total: "£300.00",
        },
        {
          title: "L.K",
          total: "£100.00",
        },
        {
          title: "H.D",
          total: "£50.00",
        },
        {
          title: "V.G",
          total: "£35.00",
        },
        {
          title: "D.C",
          total: "£20.00",
        },
        {
          title: "P.D",
          total: "£10.00",
        },
        {
          title: "L.M",
          total: "£5.00",
        },
      ],
      refresh,
      description:
        "The search for gold starts in the Sahara Desert! There’s potential for big cash prizes of up to 1,000 your stake with the Free Spins feature so you’ll want to stock up on those Story Book Free Spin Symbols to get your chance. ",
    };
    const component = renderer.create(
      <PrizeList {...data} rotate={0} onClick={() => {}} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
