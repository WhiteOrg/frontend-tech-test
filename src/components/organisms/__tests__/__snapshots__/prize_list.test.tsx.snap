// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ORGANISMS: Render testing Should render PrizeList 1`] = `
<section
  className="sc-Axmtr erLQOu"
>
  <div
    className="sc-AxmLO cVpQGk"
  >
    <h2
      className="sc-AxhUy blDKju"
      color="#091870"
    >
      Prize List
    </h2>
    <img
      className="sc-AxheI cvrYZJ"
      onClick={[Function]}
      rotate={0}
      src="refresh.svg"
    />
  </div>
  <hr
    className="sc-AxgMl ihdfMH"
  />
  <div
    className="sc-AxirZ gdibQh"
  >
    <div
      className="sc-AxiKw hrigKI"
    >
      <p
        className="sc-AxjAm DSefD"
        color="black"
        size="medium"
      >
        1
      </p>
      <p
        className="sc-AxjAm DSefD title-col"
        color="black"
        size="medium"
      >
        P.C
      </p>
      <p
        className="sc-AxjAm DSefD"
        color="black"
        size="medium"
      >
        £2,500.00
      </p>
    </div>
    <div
      className="sc-AxiKw hrigKI"
    >
      <p
        className="sc-AxjAm DSefD"
        color="black"
        size="medium"
      >
        2
      </p>
      <p
        className="sc-AxjAm DSefD title-col"
        color="black"
        size="medium"
      >
        J.S
      </p>
      <p
        className="sc-AxjAm DSefD"
        color="black"
        size="medium"
      >
        £1,250.00
      </p>
    </div>
    <div
      className="sc-AxiKw hrigKI"
    >
      <p
        className="sc-AxjAm DSefD"
        color="black"
        size="medium"
      >
        3
      </p>
      <p
        className="sc-AxjAm DSefD title-col"
        color="black"
        size="medium"
      >
        H.J
      </p>
      <p
        className="sc-AxjAm DSefD"
        color="black"
        size="medium"
      >
        £700.00
      </p>
    </div>
    <div
      className="sc-AxiKw hrigKI"
    >
      <p
        className="sc-AxjAm DSefD"
        color="black"
        size="medium"
      >
        4
      </p>
      <p
        className="sc-AxjAm DSefD title-col"
        color="black"
        size="medium"
      >
        J.V
      </p>
      <p
        className="sc-AxjAm DSefD"
        color="black"
        size="medium"
      >
        £300.00
      </p>
    </div>
    <div
      className="sc-AxiKw hrigKI"
    >
      <p
        className="sc-AxjAm DSefD"
        color="black"
        size="medium"
      >
        5
      </p>
      <p
        className="sc-AxjAm DSefD title-col"
        color="black"
        size="medium"
      >
        L.K
      </p>
      <p
        className="sc-AxjAm DSefD"
        color="black"
        size="medium"
      >
        £100.00
      </p>
    </div>
    <div
      className="sc-AxiKw hrigKI"
    >
      <p
        className="sc-AxjAm DSefD"
        color="black"
        size="medium"
      >
        6
      </p>
      <p
        className="sc-AxjAm DSefD title-col"
        color="black"
        size="medium"
      >
        H.D
      </p>
      <p
        className="sc-AxjAm DSefD"
        color="black"
        size="medium"
      >
        £50.00
      </p>
    </div>
    <div
      className="sc-AxiKw hrigKI"
    >
      <p
        className="sc-AxjAm DSefD"
        color="black"
        size="medium"
      >
        7
      </p>
      <p
        className="sc-AxjAm DSefD title-col"
        color="black"
        size="medium"
      >
        V.G
      </p>
      <p
        className="sc-AxjAm DSefD"
        color="black"
        size="medium"
      >
        £35.00
      </p>
    </div>
    <div
      className="sc-AxiKw hrigKI"
    >
      <p
        className="sc-AxjAm DSefD"
        color="black"
        size="medium"
      >
        8
      </p>
      <p
        className="sc-AxjAm DSefD title-col"
        color="black"
        size="medium"
      >
        D.C
      </p>
      <p
        className="sc-AxjAm DSefD"
        color="black"
        size="medium"
      >
        £20.00
      </p>
    </div>
    <div
      className="sc-AxiKw hrigKI"
    >
      <p
        className="sc-AxjAm DSefD"
        color="black"
        size="medium"
      >
        9
      </p>
      <p
        className="sc-AxjAm DSefD title-col"
        color="black"
        size="medium"
      >
        P.D
      </p>
      <p
        className="sc-AxjAm DSefD"
        color="black"
        size="medium"
      >
        £10.00
      </p>
    </div>
    <div
      className="sc-AxiKw hrigKI"
    >
      <p
        className="sc-AxjAm DSefD"
        color="black"
        size="medium"
      >
        10
      </p>
      <p
        className="sc-AxjAm DSefD title-col"
        color="black"
        size="medium"
      >
        L.M
      </p>
      <p
        className="sc-AxjAm DSefD"
        color="black"
        size="medium"
      >
        £5.00
      </p>
    </div>
  </div>
  <hr
    className="sc-AxgMl ihdfMH"
  />
  <p
    className="sc-AxjAm qAnnS"
    color="black"
    size="medium"
  >
    The search for gold starts in the Sahara Desert! There’s potential for big cash prizes of up to 1,000 your stake with the Free Spins feature so you’ll want to stock up on those Story Book Free Spin Symbols to get your chance. 
  </p>
</section>
`;
